jinahub/indexers/searcher/FaissSearcher/faiss_searcher.py

```diff
@@ -108,10 +109,15 @@ def __init__(
 
         self._doc_ids = []
         self._doc_id_to_offset = {}
+        self._is_deleted = []
         self._prefetch_data = []
 
         self.logger = get_logger(self)
-        self._load_dump(dump_path, dump_func, prefetch_size, **kwargs)
+        is_loaded = False
+        if os.path.exists(self.snapshot_path):
```

Comment:
When is `self.snapshot` assigned/created?

Reply:
As the endpoint `def snapshot()` is called, the snapshot folder will be created (or overrided).